vms/proposervm/post_fork_block.go

```diff
@@ -128,6 +136,41 @@ func (b *postForkBlock) verifyPostForkChild(ctx context.Context, child *postFork
 	)
 }
 
+func (b *postForkBlock) verifyVRFSig(ctx context.Context, child *postForkBlock, parentPChainHeight uint64) error {
+	if !b.vm.Config.IsEActivated(child.Timestamp()) {
+		// if the VRFSig has yet to be activated, verify that the VRF signature isn't there.
+		if len(child.SignedBlock.VRFSig()) != 0 {
+			return errVRFSignaturePresents
+		}
+		return nil
+	}
+
+	var proposerPublicKey *bls.PublicKey
+	// find out the proposer for this block.
+	// if there is a known proposer. ( i.e. block was signed )
+	if childProposer := child.SignedBlock.Proposer(); childProposer != ids.EmptyNodeID {
+		// get the validators set.
+		validatorSet, err := child.vm.ctx.ValidatorState.GetValidatorSet(ctx, parentPChainHeight, b.vm.ctx.SubnetID)
+		if err != nil {
+			return err
+		}
+
+		// get the validator for this proposer
+		childValidator := validatorSet[childProposer]
```

Comment:
nit: this is fine, but I believe usually we use the boolean to check whether a key is present in a map, rather than checking whether the value is not nil

Reply:
let me do both, just to be on the safe side ?